fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersUtil.java

```diff
@@ -79,185 +64,95 @@
-            return populateSearchParameterMapFromStream(stream);
+            map = buildSearchParametersMapFromBundle(bundle);
+        } catch (Exception e) {
+            log.log(Level.SEVERE, "Unexpected error while loading built-in search parameters", e);
+            map = Collections.emptyMap();
         }
-    }
 
-    /**
-     * populates the search parameter map and defaults to json
-     * 
-     * @param stream
-     * @return
-     */
-    public static Map<String, Map<String, SearchParameter>> populateSearchParameterMapFromStream(InputStream stream) {
-        return populateSearchParameterMapFromStreamByFormat(stream, Format.JSON);
+        return map;
     }
 
     /**
-     * populates the search parameter map and is XML
+     * Builds a Map of ParameterMaps from the passed Bundle.
      * 
-     * @param stream
-     * @return
-     */
-    public static Map<String, Map<String, SearchParameter>> populateSearchParameterMapFromStreamXML(InputStream stream) {
-        return populateSearchParameterMapFromStreamByFormat(stream, Format.XML);
-    }
-
-    /*
-     * Loads SearchParameters using the specified InputStream and returns a Map containing them.
-     * @param stream the InputStream from which to load the SearchParameters
-     * @param Format
-     * @return
+     * @param bundle a Bundle of type Collection with entries of type SearchParameter
+     * @return a Map of ParameterMaps, keyed by resourceType
+     * @throws ClassCastException if the Bundle contains entries of any type other than SearchParameter
      */
-    private static Map<String, Map<String, SearchParameter>> populateSearchParameterMapFromStreamByFormat(InputStream stream, Format format) {
-        // Format is never null, as this method is private and hidden.
-        // In order to maintain this contract, and avoid the check, keep private.
-
-        Map<String, Map<String, SearchParameter>> searchParameterMap = new HashMap<>();
-
-        // The code block is isolating the effects of exceptions by capturing the exceptions here
-        // and returning an empty searchParameterMap, and continuing operation.
-        // The failure is logged out.
-        try {
-            // The code is agnostic to format.
-            Bundle bundle = FHIRParser.parser(format).parse(stream);
-
-            FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
-            EvaluationContext evaluationContext = new EvaluationContext(bundle);
-            Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, FHIR_PATH_BUNDLE_ENTRY);
-
-            List<SearchParameter> paramList = result.stream()
-                    .map(node -> node.asResourceNode().resource().as(SearchParameter.class))
-                    .collect(Collectors.toList());
-            for (SearchParameter parameter : paramList) {
+    public static Map<String, ParametersMap> buildSearchParametersMapFromBundle(Bundle bundle) {
```

Comment:
any reason for the separation of bundle vs InputStream? curiosity not critique ... 
this is technically now a breaking change. since 
`Map<String, Map<String, SearchParameter>> populateSearchParameterMapFromStreamXML(InputStream stream) {` and the corresponding json method are now public in 4.0.0 

Yes, I highly doubt someone is using this exact method, however, we should carefully treat these method changes going forward.  Once again - discussion point not a critique

Reply:
Yeah, not sure how careful to be about preserving backwards compatibility on the public methods...this is one reason i was pushing to better define internal vs external api.

In this particular case, I just thought it was a little cleaner to the map building logic from the reading of the Bundle from disk.  It means we don't need to deal with the JSON vs XML variants or deal with "what if the passed file isn't a Bundle" or what-have-you...basically just a tighter API.

Originally, I had a variant that took a list of SearchParameters instead of a Bundle (since thats all we really need as input), but I ended up removing that since we're already expecting to have a Bundle in the only two places calling this method right now.

Anyway, I'm not dead-set on this particular change (InputStream -> Bundle), so let me know if you'd like to see it reverted in favor of backwards compat.

Reply:
No change needed at all.  Just thinking ahead.